doc/sphinxman/source/add_tests.rst

```diff
@@ -30,108 +30,312 @@
 
 .. _`faq:add_tests`:
 
+==========
+Test Suite
+==========
+
+A test suite plays a vital role in open-source software use and development.
+
+* For a |PSIfour| user, tests provide models of inputs that should work
+  "as-is" and a searchable collection of syntax and capabilities.
+  The test suite also allows high-quality development snapshots of the
+  codebase to be built automatically for users.
+
+* For a CMS user, a test suite alongside installed |PSIfour| can be used to show that this piece is working in a complex software environment.
+
+* For a feature developer, adding tests provides confidence that you
+  can leave your code untouched and still advertise that the feature works
+  years later. With tests, proposed changes to |PSIfour| that break your
+  code fall upon the change proposer to fix, rather than being merged
+  silently and lying in wait for a concientious user to detect and report
+  and then likely falling upon *you* to fix.
+
+* For a general developer, the test suite allows confidence in refactoring, switching out underlying libraries, maintenance, and upgrading.
+
+
+CTest and pytest, PSIthon and PsiAPI
+====================================
+
+In designing a test, sometimes you want it to be a model input for the user in a single file or you don't want a lot of ``psi4.`` or Python syntax cluttering the input.
+In this case, follow :ref:`faq:add_psithon_tests` to prepare as PSIthon (``psi4 input.dat``) for, roughly speaking, running through :program:`ctest`.
+The PSIthon/CTest test suite occupies the whole of :source:`tests` *except* :source:`tests/pytests`.
+
+At other times you want the test to check several variations of a template job or you want to test error handling or you want to focus on PsiAPI rather than PSIthon or you want to control the compute conditions with environment variables.
+In this case, follow :ref:`faq:add_psiapi_tests` to prepare as PsiAPI (``import psi4``) for, roughly speaking, running through :program:`pytest`.
+The PsiAPI/pytest test suite occupies :source:`tests/pytests`.
+
+The above description sounds as if there are two disjoint test suites, and you have to run both ``ctest`` and ``pytest`` to fully test |PSIfour|.
+This has indeed been the case until March 2022.
+The difficulty has been that (1) two test suites is unexpected so some developers don't know to run both; and (2) there are important tests in the PSIthon suite that can't be run on a |PSIfour| installation since CTest only works in a build directory.
```

Comment:
Does the Sphinx support footnotes? If so, this line should be one. It's important for experienced developers who want to know why things changed, but for new developers, it interrupts the line of thought.

Reply:
Yes, Sphinx can do footnotes, but I'd rather keep this bit all together. The appearance of two suites isn't going away directory-structure-wise, so may as well face them.

(New developers should be happy I didn't emphasize that in the source, the ctest dir contains the pytest dir, while in the installation, the pytest dir contains the ctests dirs. :-)

Reply:
I completely agree with discussing "it looks like there are two test suites" here. But the specific thing that I think should be footnoted is the impetus for making ctests accessible through pytest.

Reply:
This is the one item I have left on this PR. I can agree to disagree and approve regardless, but I stand by my previous comment. Whether this particular line is footnoted is unrelated to facing the appearance of two test suites.

Reply:
Yeah, I played with that paragraph some. I agree it can read cleaner w/o the 2-point sentence, but I didn't find a better place for the sentence. And footnote looks odd when whole docs section is 3 vertical inches. I'm inclined to leave it for now.